.circleci/config/commands/install-vault.yml

```diff
@@ -2,10 +2,28 @@ parameters:
   version:
     type: string
     default: 1.2.4
+  os:
+    type: string
+    default: linux
 steps:
   - run:
-      name: Install Vault << parameters.version >>
+      name: Install Vault
       command: |
-        curl -SL --fail -o /tmp/vault.zip https://releases.hashicorp.com/vault/<< parameters.version >>/vault_<< parameters.version>>_linux_amd64.zip
-        sudo unzip -d /usr/local/bin /tmp/vault.zip
-        rm -rf /tmp/vault*
+        set -x
+
+        curl -SL --fail -o /tmp/vault.zip https://releases.hashicorp.com/vault/"<< parameters.version >>"/vault_"<< parameters.version >>"_"<< parameters.os >>"_amd64.zip
+
+        if [[ << parameters.os >> == "linux" ]]
+        then
+          sudo unzip -d /usr/local/bin /tmp/vault.zip
+          rm -rf /tmp/vault*
+        fi
+        if [[ << parameters.os >> == "windows" ]]
+        then
+          if [[ ! -f "c:\Windows\vault.exe" ]]
```

Comment:
I like that we're testing that we have an installation here, but we've already done the work of downloading Vault at this point. Could we improve this (and include the check for Linux too) by checking that Vault exists and that it's the version we expect, and then download and install if the checks don't pass?

Looks like we could do the same for the other installation targets too.

Reply:
I don't know Windows, but is it common to store 3rd party apps in `c:\windows`?  I assume this is a matter of convenience as it's already in `PATH` env var?